### PR TITLE
First pass PR to polish the mock-builder example

### PR DESCRIPTION
**Pull request by lemunozm**

### CODE REVIEW

File: substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+///         fn unreserve(a: &T::AccountId, b: Self::Balance) -> Self::Balance;
+///         fn repatriate_reserved(
+///             a: &T::AccountId,
+///             b: &T::AccountId,
+///             c: Self::Balance,
+///             d: BalanceStatus,
+///         ) -> Result<Self::Balance, DispatchError>;
+///     }
+/// }
+/// ```
```
Comment (lemunozm) [inline substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs]: I'm not sure if this is a selling point or could create more confusing given the number of methods that should be mocked. Just wanted to show a complex example.

File: substrate/frame/examples/mock-builder/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		pub fn create_auction(origin: OriginFor<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// Check reserve
+			let current = T::Currency::reserved_balance(&who);
+			ensure!(current >= T::ExpectedAmount::get(), Error::<T>::NotEnoughDeposit);
+
+			// Check time
+			let now = T::Time::now();
+			let ready_at = T::WaitingTime::get() + LastDeposit::<T>::get(who).unwrap_or(now);
```
Comment (wischli) [inline substrate/frame/examples/mock-builder/src/lib.rs]: Even if this a "silly" example, let's use `ensure_add` or `saturating_add`.

File: substrate/frame/examples/mock-builder/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+};
+use frame_system::RawOrigin;
+
+const ALICE: u64 = 2;
+
+#[test]
+fn reserve() {
+	new_test_ext().execute_with(|| {
+		// Mock the internal call done to T::Currency::reserve() inside of
+		// MyPaller:make_reserve() checking the expected inputs and returining a successfull
```
Comment (wischli) [inline substrate/frame/examples/mock-builder/src/tests.rs]:
```suggestion
		// MyPallet:make_reserve() checking the expected inputs and returning a successful
```

File: substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+		fn now() -> Self::Moment {
+			execute_call!(())
+		}
+	}
+}
+
+/// With procedural support, it will be autogenerated as follows,
+/// implementing all mock_* functions.
+///
+/// Be aware that this show a very complex use case with several traits with a bunch of functions
```
Comment (wischli) [inline substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs]:
```suggestion
/// Be aware that this shows a very complex use case with several traits with a bunch of functions
```

File: substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs (hunk 1 above)
Comment (wischli) [reply substrate/frame/examples/mock-builder/src/mock/mock_pallets.rs]: I think it causes confusion but then again I am also not favoring removing it. Let's keep it.

**Review by mustermeiszer — COMMENTED** [review]
@lemunozm this looks good. Could you open a PR on the `polkadot-sdk` repo from parity?

**Review by wischli — APPROVED** [review]
Thanks a lot for all the additional `mock-builder` work! I have a few minor nits which can be ignored. We are good to open this in on upstream!

File: substrate/frame/examples/mock-builder/src/lib.rs (hunk 2 above)
Comment (lemunozm) [reply substrate/frame/examples/mock-builder/src/lib.rs]: Aggre

**Pull request closed without merging**